File: android/versioned-abis/expoview-abi49_0_0/src/main/java/abi49_0_0/host/exp/exponent/modules/api/components/reactnativestripesdk/addresssheet/AddressSheetViewManager.kt
package abi49_0_0.host.exp.exponent.modules.api.components.reactnativestripesdk.addresssheet

import abi49_0_0.com.facebook.react.bridge.ReadableArray
import abi49_0_0.com.facebook.react.bridge.ReadableMap
import abi49_0_0.com.facebook.react.common.MapBuilder
import abi49_0_0.com.facebook.react.uimanager.SimpleViewManager
import abi49_0_0.com.facebook.react.uimanager.ThemedReactContext
import abi49_0_0.com.facebook.react.uimanager.annotations.ReactProp

class AddressSheetViewManager : SimpleViewManager<AddressSheetView>() {
  override fun getName() = "AddressSheetView"

  override fun getExportedCustomDirectEventTypeConstants(): MutableMap<String, Any> {
    return MapBuilder.of(
      AddressSheetEvent.ON_SUBMIT, MapBuilder.of("registrationName", "onSubmitAction"),
      AddressSheetEvent.ON_ERROR, MapBuilder.of("registrationName", "onErrorAction")
    )
  }

  @ReactProp(name = "visible")
  fun setVisible(view: AddressSheetView, visibility: Boolean) {
    view.setVisible(visibility)
  }

  @ReactProp(name = "appearance")
  fun setAppearance(view: AddressSheetView, appearance: ReadableMap) {
    view.setAppearance(appearance)
  }

  @ReactProp(name = "defaultValues")
  fun setDefaultValues(view: AddressSheetView, defaults: ReadableMap) {
    view.setDefaultValues(defaults)
  }

  @ReactProp(name = "additionalFields")
  fun setAdditionalFields(view: AddressSheetView, fields: ReadableMap) {
    view.setAdditionalFields(fields)
  }

  @ReactProp(name = "allowedCountries")
  fun setAllowedCountries(view: AddressSheetView, countries: ReadableArray) {
    view.setAllowedCountries(countries.toArrayList().filterIsInstance<String>())
  }

  @ReactProp(name = "autocompleteCountries")
  fun setAutocompleteCountries(view: AddressSheetView, countries: ReadableArray) {
    view.setAutocompleteCountries(countries.toArrayList().filterIsInstance<String>())
  }

  @ReactProp(name = "primaryButtonTitle")
  fun setPrimaryButtonTitle(view: AddressSheetView, title: String) {
    view.setPrimaryButtonTitle(title)
  }

  @ReactProp(name = "sheetTitle")
  fun setSheetTitle(view: AddressSheetView, title: String) {
    view.setSheetTitle(title)
  }

  @ReactProp(name = "googlePlacesApiKey")
  fun setGooglePlacesApiKey(view: AddressSheetView, key: String) {
    view.setGooglePlacesApiKey(key)
  }

  override fun createViewInstance(reactContext: ThemedReactContext): AddressSheetView {
    return AddressSheetView(reactContext)
  }
}

File: ios/vendored/sdk48/@stripe/stripe-react-native/ios/AddressSheet/AddressSheetViewManager.m
//
//  AddressSheetViewManager.m
//  stripe-react-native
//
//  Created by Charles Cruzan on 10/11/22.
//

#import <Foundation/Foundation.h>
#import <ABI48_0_0React/ABI48_0_0RCTBridgeModule.h>
#import <ABI48_0_0React/ABI48_0_0RCTViewManager.h>

@interface ABI48_0_0RCT_EXTERN_REMAP_MODULE(AddressSheetViewManager, ABI48_0_0AddressSheetViewManager, ABI48_0_0RCTViewManager)
ABI48_0_0RCT_EXPORT_VIEW_PROPERTY(visible, BOOL)
ABI48_0_0RCT_EXPORT_VIEW_PROPERTY(presentationStyle, NSString)
ABI48_0_0RCT_EXPORT_VIEW_PROPERTY(animationStyle, NSString)
ABI48_0_0RCT_EXPORT_VIEW_PROPERTY(appearance, NSDictionary)
ABI48_0_0RCT_EXPORT_VIEW_PROPERTY(defaultValues, NSDictionary)
ABI48_0_0RCT_EXPORT_VIEW_PROPERTY(additionalFields, NSDictionary)
ABI48_0_0RCT_EXPORT_VIEW_PROPERTY(allowedCountries, NSArray)
ABI48_0_0RCT_EXPORT_VIEW_PROPERTY(autocompleteCountries, NSArray)
ABI48_0_0RCT_EXPORT_VIEW_PROPERTY(primaryButtonTitle, NSString)
ABI48_0_0RCT_EXPORT_VIEW_PROPERTY(sheetTitle, NSString)
ABI48_0_0RCT_EXPORT_VIEW_PROPERTY(onSubmitAction, ABI48_0_0RCTDirectEventBlock)
ABI48_0_0RCT_EXPORT_VIEW_PROPERTY(onErrorAction, ABI48_0_0RCTDirectEventBlock)
@end
